designer/server/src/main/scala/pl/touk/nussknacker/ui/process/deployment/DeploymentServiceImpl.scala

```diff
@@ -336,6 +343,39 @@ class DeploymentServiceImpl(
     } yield result)
   }
 
+  override def fetchProcessStatesForProcesses(processes: List[BaseProcessDetails[Unit]])(
+      implicit user: LoggedUser,
+      ec: ExecutionContext,
+      freshnessPolicy: DataFreshnessPolicy
+  ): Future[Map[String, ProcessState]] =
+    for {
+      processesInProgress <- getInProgressActionTypesForAllProcesses
+      processStatus <- processes
+        .map(process => Future.successful(process.name) zip getProcessState(process, processesInProgress))
+        .sequence
+    } yield processStatus.toMap
+
+  override def enrichDetailsWithProcessState(processList: List[BaseProcessDetails[_]])(
+      implicit user: LoggedUser,
+      ec: ExecutionContext,
+      freshnessPolicy: DataFreshnessPolicy
+  ): Future[List[BaseProcessDetails[_]]] =
+    for {
+      processesInProgress <- getInProgressActionTypesForAllProcesses
+      processesWithState <- processList.map {
+        case process if process.isFragment => Future.successful((process, None))
+        case process => Future.successful(process) zip getProcessState(process, processesInProgress).map(Option.apply)
```

Comment:
You can use `.map(Option)` or `.map(Option(_))`. Let's try to avoid explicitly calling `apply`